V4.0.9-Xentraxx

Xentraxx opened the pull request:
Heyho Wachee! Would love it if you could create a prerelease version of 4.0.9.
I need some more exposure/people to test it to get some feedback and fix possible bugs before I release it.

Xentraxx committed:
fixed hex encoded emojis in json

Xentraxx committed:
enabled write exif by default

Xentraxx committed:
using native windows shortcut creation again

Xentraxx committed:
added bye bye comment

Xentraxx committed:
Merge branch 'prerelease'

Xentraxx committed:
typo

Xentraxx committed:
year folder fix

Xentraxx committed:
fixed issue #25

https://github.com/Xentraxx/GooglePhotosTakeoutHelper/issues/25

Only folders beginning with "Photos from ..." should be deemed year folders

Xentraxx committed:
fixed tests after isYearAlbum change

Xentraxx committed:
clarified default behaviour in docu

Xentraxx committed:
added edge case handling for AVI videos

Xentraxx committed:
skipping AVI exif writes

Xentraxx committed:
changelog and version update

Xentraxx committed:
skipping AVI gps exif write

Xentraxx committed:
removed utf8 arg in exiftool

Xentraxx committed:
removed modify json mode

Xentraxx committed:
making lint happy

Xentraxx committed:
implemented --fix-extensions mode

Xentraxx committed:
removed unnecessary test

Xentraxx committed:
updated readme

Xentraxx committed:
changed error to warning

https://github.com/Xentraxx/GooglePhotosTakeoutHelper/pull/21#issuecomment-2961144875

Xentraxx committed:
using jsonForFile() function in fixIncorrectExtensions method to find json

Xentraxx committed:
added tryhard usage to find json in extensionFix logic

Xentraxx committed:
removePartialExtraFormats fix

addresses issue #29

Xentraxx committed:
updated changelog

Xentraxx committed:
added logging

Xentraxx committed:
added some unit tests

Xentraxx committed:
revived interactive unzipping

Xentraxx committed:
fixed mp4 to AVI extension fixing

Xentraxx committed:
removed unnecessary info in changelog

Xentraxx committed:
experimental refactoring Phase 1

Xentraxx committed:
experimental refactoring Phase 2

Should be a stable version

Xentraxx committed:
experimental refactoring phase 2 fix

Xentraxx committed:
fix and restructuring of truncated file names

Xentraxx committed:
Merge branch 'prerelease' into v4.0.9-development

Xentraxx committed:
experimental refactoring phase 3

Xentraxx committed:
experimental refactoring phase 3 fix

Xentraxx committed:
experimental refactoring phase 3 fixes

Xentraxx committed:
renamed files to fit new structure

Xentraxx committed:
experimental refactoring phase 3 final fixes

Xentraxx committed:
fixed duration formatting

Xentraxx committed:
experimental moving refactoring

Xentraxx committed:
performance improvements

Xentraxx committed:
removed removeDigitStrategy()

Xentraxx committed:
fixed truncated supplemental metadata

Xentraxx committed:
fix order naming

Xentraxx committed:
fix json finder

Xentraxx committed:
removed temporary tests and debug files

Xentraxx committed:
speeded up ExifTool 10x-50x

Xentraxx committed:
massive performance improvements

Xentraxx committed:
linting fix

Xentraxx committed:
dart format

Xentraxx committed:
experimental refactoring phase 4

Xentraxx committed:
bug fixes

Xentraxx committed:
experimental refactoring phase 4.2

Xentraxx committed:
refactoring phase 4.3

Xentraxx committed:
Refactoring phase 4.4

Xentraxx committed:
refactoring phase 4.5

Xentraxx committed:
refactoring 4.6

Xentraxx committed:
refactoring phase 4.7 (phase 4 done)

Xentraxx committed:
refactoring phase 5.1

Xentraxx committed:
refactoring phase 5.2

Xentraxx committed:
refactoring phase 5.3

Xentraxx committed:
refactoring phase 5.4

Xentraxx committed:
refactoring phase 5.5

Xentraxx committed:
added tests

Xentraxx committed:
test fixes

Xentraxx committed:
added more tests and did some fixes

Xentraxx committed:
fixed tests

Xentraxx committed:
print fixes

Xentraxx committed:
unzipping fixes

Xentraxx committed:
ui improvements

Xentraxx committed:
presentation improvements

Xentraxx committed:
interactive mode fixes

Xentraxx committed:
interactive mode cleanup

Xentraxx committed:
fixed invalid answers

Xentraxx committed:
finished fixing interactive mode

Xentraxx committed:
fix output

Xentraxx committed:
documentation updates

Xentraxx committed:
implementation of all moving modes

Xentraxx committed:
cleanup of unused methods

Xentraxx committed:
added e2e tests

Xentraxx committed:
step 8 output fix

Xentraxx committed:
reorganized tests

Xentraxx committed:
test fixes

Xentraxx committed:
Fixed album creation logic

Xentraxx committed:
moving logic fix and extended test coverage

Xentraxx committed:
improved unix free space detection with fallbacks

Xentraxx committed:
log fix and removal of deprecated functionality

Xentraxx committed:
improved step 1 logging

Xentraxx committed:
fixed duplicate detection logging

Xentraxx committed:
fix step 3 logging

Xentraxx committed:
fix logging step 4

Xentraxx committed:
coordinates writing to exif fix

Xentraxx committed:
coordinate test fix

Xentraxx committed:
removed redundant test files

Xentraxx committed:
fixed crash through unsupported AVI

Xentraxx committed:
added caching to hash calculation

Xentraxx committed:
enhanced final output

Xentraxx committed:
changelog update

Xentraxx committed:
removed deprecated function

Xentraxx committed:
removed unnecessary abstraction layer

Xentraxx committed:
structure refactoring

Xentraxx committed:
utils.dart removal

Xentraxx committed:
added tests for issues

Xentraxx committed:
test fixes

Xentraxx committed:
updated note

Xentraxx committed:
added issue template

Xentraxx committed:
fixed issue template

Xentraxx committed:
nothing mode fix (making sure no pictures from albums are lost in nothing mode

Xentraxx committed:
fixed test expectations

Xentraxx committed:
added .aicontext

Xentraxx committed:
extended context

Xentraxx committed:
refactoring

Xentraxx committed:
fixed tests

Xentraxx committed:
removal of copy mode references

Xentraxx committed:
removal of deprecated stuff

Xentraxx committed:
Legacy code removal and cleanup

Xentraxx committed:
refactoring

Xentraxx committed:
removal of deprecated service

Xentraxx committed:
performance improvgements

Xentraxx committed:
race condition fix

Xentraxx committed:
fixed bugs and failing e2e tests

Xentraxx committed:
added exif validation

Xentraxx committed:
fixed failing tests

Xentraxx committed:
formatting

Xentraxx committed:
fixed file synchronization race conditions

Xentraxx committed:
making lint happy

Xentraxx committed:
Fixed Issue #39

Xentraxx committed:
lint fix

Xentraxx committed:
fixed changelog

Xentraxx committed:
Update CHANGELOG.md

Xentraxx committed:
Update README.md

jaimetur commented:
Great. Would be good to include a PR also in the original repository from TheLastGimbus which is where more exposure it will have

Xentraxx commented:
TheLastGimbus saw my prior pull request but also said he focuses on stability and in the future might look at it.

Xentraxx committed:
Update README.md

https://github.com/TheLastGimbus/GooglePhotosTakeoutHelper/issues/392

Xentraxx committed:
removed unnecessary wrapper file

Xentraxx committed:
unified logging service

Xentraxx committed:
Fix Issue #45

Xentraxx committed:
fix issue #47

Xentraxx committed:
issue #46 fix and extended logging

Xentraxx committed:
Issue #44 Step 5 30% improvement

Step 5 - Write EXIF Data:
Sequential EXIF writing: 2.02 seconds
Parallel EXIF writing: 1.56 seconds
Performance gain: ~30% faster (0.46 seconds improvement)

Xentraxx committed:
Improved write-exif performance when fixing extensions

When fixing extensions we don't need to read mimetype from header again when we actually write to exif because we assume that extensions have been fixed and therefore just look at the path of the file and determine mimetype from there, reducing read operations.

Xentraxx committed:
Revert "Improved write-exif performance when fixing extensions"

This reverts commit 5899fd8582cf201d1311224a6dbc0cc859e9b230.

Xentraxx committed:
moved method only used in tests to test file

Xentraxx committed:
fix for scenario where exiftool is not installed

EXIF extractor was only included when ExifTool was available, which meant that files couldn't have their EXIF data extracted using native methods when ExifTool wasn't installed.

Xentraxx committed:
added panasonic raw support

Xentraxx committed:
fix for issue #50

changed print to logdebug

Xentraxx committed:
Fix Issue #50

Symlinks instead of shortcuts. Still some failing tests. Need to investigate further

Xentraxx committed:
fixed symlink detection in tests

Xentraxx committed:
changed readme

Xentraxx commented:
4.1.0 prerelease is out: https://github.com/Xentraxx/GooglePhotosTakeoutHelper/actions/runs/16541824034

Update:
This is old already again. We're at 5.0.5 currently.
